File: sdk/ts/src/examples/basic.ts

import { M0Client } from "../client";

async function main() {
  const client = new M0Client({
    baseUrl: process.env.M0_API_BASE ?? "http://localhost:8080"
  });

  const health = await client.health();
  console.log("health:", health);

  const markets = await client.listMarkets();
  console.log("markets:", markets);

  if (markets.length > 0) {
    const p = await client.latestPrediction(markets[0].market_id);
    console.log("latest:", p);
  }
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});

File: sdk/rust/src/examples/basic.rs

use crate::M0Client;

pub async fn run() -> anyhow::Result<()> {
    let base = std::env::var("M0_API_BASE").unwrap_or_else(|_| "http://localhost:8080".into());
    let client = M0Client::new(&base)?;

    let health = client.health().await?;
    println!("health: {}", health);

    let markets = client.markets().await?;
    println!("markets: {}", serde_json::to_string_pretty(&markets)?);

    if let Some(m) = markets.first() {
        let p = client.latest_prediction(&m.market_id).await?;
        println!("latest: {}", serde_json::to_string_pretty(&p)?);
    }

    Ok(())
}
